src/Features/VisualBasic/Portable/Completion/CompletionProviders/CompletionUtilities.vb

```diff
@@ -123,10 +123,19 @@ Namespace Microsoft.CodeAnalysis.VisualBasic.Completion.Providers
                 End If
             End If
 
-            ' If the user is attempting to escape something, escape the item if it isn't already
-            ' escaped
-            If ch = "]"c AndAlso insertionText(0) <> "["c Then
-                Return "[" + insertionText
+            If ch = "]"c Then
+                If insertionText(0) <> "["c Then
+                    ' user is committing with ].  If the item doesn't start with '['
+                    ' then add that to the beginning so [ and ] properly pair up.
+                    Return "[" + insertionText
```

Comment:
> I guess I'm wondering why both Ifs can't be true at the same time

Because we never generate anything that doesn't start with a [ but does end with a ].